src/compiler/checker.ts

```diff
@@ -44446,77 +44452,125 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         else {
             checkTypeAssignableTo(checkExpression(initializer), numberType, initializer, Diagnostics.Type_0_is_not_assignable_to_type_1_as_required_for_computed_enum_member_values);
         }
-        return value;
+        return undefined;
     }
 
-    function evaluate(expr: Expression, location?: Declaration): string | number | undefined {
+    function evaluate(expr: Expression, location?: Declaration): { value: string | number | bigint | boolean | null | undefined; } | undefined {

```

Comment:
I'm open to using a different strategy between differentiating evaluated `undefined` and lack of the evaluation result.

Reply:
Could always use a Symbol as a sentinel value.

Reply:
This actually crossed my mind :D but I wasn't sure if this should just be `return Symbol()` or if I should introduce some global symbol or what. Code style - I prefer the current approach but it might be nice to avoid introducing super disposable objects like this in the compiler.

Reply:
Disposable objects is what I'm thinking is going to bite this; I'd just make a const local or something. But maybe it doesn't actually matter.

Reply:
I doubt it matters so is fine to leave as-is; we have worse things like helpers which almost always return empty arrays of 1 item (boo)